src/api/search.js

```diff
@@ -1,31 +1,39 @@
 import { cachedFetch, buildMwApiUrl } from 'utils'
 
 export const search = (lang, term) => {
+  // fulltext search
   const params = {
     action: 'query',
-    prop: ['description', 'pageimages', 'pageprops'].join('|'),
+    prop: 'pageimages',
+    generator: 'search',
+    list: 'search',
+    srprop: 'snippet',
+    srsearch: term,
+    srlimit: 15,
+    srenablerewrites: true,
+    srinfo: 'rewrittenquery',
+    gsrsearch: term,
+    gsrnamespace: 0,
+    gsrlimit: 15,
     piprop: 'thumbnail',
-    pilimit: 15,
-    ppprop: 'displaytitle',
-    generator: 'prefixsearch',
-    redirects: true,
     pithumbsize: 64,
-    gpslimit: 15,
-    gpsnamespace: 0,
-    gpssearch: term.replace(/:/g, ' ')
+    pilimit: 15,
+    format: 'json'
   }
+
   const url = buildMwApiUrl(lang, params)
   return cachedFetch(url, data => {
-    if (!data.query || !data.query.pages) {
+    if (!data.query || !data.query.search) {
       return []
     }
-    data.query.pages.sort((a, b) => a.index - b.index)
-    return Object.values(data.query.pages).map((p) => {
+
+    return Object.values(data.query.search).map((p) => {
+      const page = data.query.pages.find(page => page.pageid === p.pageid)
       return {
         title: p.title,
-        titleHtml: p.title.replace(term, `<span class="searchmatch">${term}</span>`),
-        description: p.description,
-        imageUrl: p.thumbnail && p.thumbnail.source
+        titleHtml: p.title,
+        description: p.snippet.replace(/<span class="searchmatch">(\w+)<\/span>/g, '$1'),
```

Comment:
so we change to fulltext search, it returns `snippet` instead of the article `description`, to be confirmed that whether we need to highlight the searchterm in snippet.